### PR TITLE
UPDATED: Add config option to enable the encryption of AWS EKS secrets

### PR DESCRIPTION
**Pull request by joneszc**
## Reference Issues or PRs
Fixes [#2681](https://github.com/nebari-dev/nebari/issues/2681)
Fixes [#2746](https://github.com/nebari-dev/nebari/issues/2746)
Modifies [PR#2723](https://github.com/nebari-dev/nebari/pull/2723)

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### Allows user to set EKS encryption of secrets by specifying a KMS key ARN in nebari-config.yaml
```
amazon_web_services:
  eks_kms_arn: 'arn:aws:kms:us-east-1:010101010:key/3xxxxxxx-xxxxx-xxxxx-xxxxx'
```
image

### The KMS key must meet the following conditions:
* Symmetric
* Can encrypt and decrypt data
* Created in the same AWS Region as the cluster
* If the KMS key was created in a different account, the [IAM principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html#iam-term-principal) must have access to the KMS key.

### CODE REVIEW

**Review by dcmcand — CHANGES_REQUESTED** [review]
@joneszc , please add a mock for `amazon_web_services.kms_key_arns` to https://github.com/nebari-dev/nebari/blob/215bbd622f0ea1692e79bb4411db81ef28b1d329/tests/tests_unit/conftest.py#L58 to allow unit tests to pass without needing to hit actual AWS endpoints.

File: src/_nebari/provider/cloud/amazon_web_services.py [hunk 1]
```diff
@@ -121,6 +121,30 @@ def instances(region: str) -> Dict[str, str]:
...
+    fields = [
+        "Arn",
+        "KeyUsage",
+        "KeySpec",
+        # "KeyState",
+        # "Origin",
+        # "KeyManager",
+        # "EncryptionAlgorithms",
+        # "MultiRegion",
+    ]
```
Comment (viniciusdc) [inline src/_nebari/provider/cloud/amazon_web_services.py:139]: I would remove the extra keys from the request, and add a comment with a link to the python API docs with the extra available fields instead.

File: src/_nebari/provider/cloud/amazon_web_services.py [hunk 2]
```diff
@@ -121,6 +121,30 @@ def instances(region: str) -> Dict[str, str]:
...
+    kms_keys = [
+        client.describe_key(KeyId=j["KeyId"]).get("KeyMetadata")
+        for i in paginator.paginate()
+        for j in i["Keys"]
+    ]
+    return {i["KeyId"]: {k: i[k] for k in fields} for i in kms_keys if i["Enabled"]}
```
Comment (viniciusdc) [inline src/_nebari/provider/cloud/amazon_web_services.py:145]: Could you add a comment at the beginning with a link to the docs for future reference? On the response object schema: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kms/client/describe_key.html

**Review by viniciusdc — CHANGES_REQUESTED** [review]

**Comment by viniciusdc** [comment]
Looks great! thanks @joneszc. I would ask now just to retrigger the tests and if possible, if you can do an AWS deployment  with the following:
- deploy a fresh version of nebari from this branch without any modification to the yaml file;
- add the extra field above and redeploy
- if the step before succeeded, try removing it and checking the behavior of the cluster during terraform application. 

This requires a subsequent docs PR, if it already exists, can you link here as well?

File: src/_nebari/provider/cloud/amazon_web_services.py (hunk 1 above)
Comment (joneszc) [reply src/_nebari/provider/cloud/amazon_web_services.py:139]: Copy.  I'll implement this in [PR#2788](https://github.com/nebari-dev/nebari/pull/2788)

File: src/_nebari/provider/cloud/amazon_web_services.py (hunk 2 above)
Comment (joneszc) [reply src/_nebari/provider/cloud/amazon_web_services.py:145]: Yes, I'll implement this in [PR#2788](https://github.com/nebari-dev/nebari/pull/2788)

**Pull request closed without merging**